### PR TITLE
Add container instances

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Adds container instance modules

### What issues does this PR fix or reference?
Fixes: #30 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to idem-azurerm require tests.**
- [x] Docs
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [idem-azurerm's Contributing Guide](https://github.com/eitrtechnologies/idem-azurerm/blob/master/.github/CONTRIBUTING.md) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: idem_azurerm/exec/azurerm/containerinstance/group.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+
+            **timeout_seconds**: The timeout seconds.
+
+    :param os_type: The operating system type required by the containers in the container group. Possible values
+        include: 'Windows', 'Linux'
+
+    :param restart_policy: Restart policy for all containers within the container group. Possible values are:
+    - ``Always``: Always restart
+    - ``OnFailure``: Restart on failure
+    - ``Never``: Never restart.
```
Comment (Ajnbro) [inline idem_azurerm/exec/azurerm/containerinstance/group.py:172]: Remove this period for uniformity

File: idem_azurerm/exec/azurerm/containerinstance/group.py [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+        container group.
+
+    :param image_registry_credentials: A list of dictionaries defining ImageRegistryCredential objects for the image
+        registry credentials.
+
+    :param ip_address: A dictionary defining an IpAddress object which represents the IP address for the container
+        group. Possible keys are:
+    - ``ports``: Required if ip_address is used. The list of ports exposed on the container group.
+    - ``type``: Required if ip_address is used. Specifies if the IP is exposed to the public internet or private VNET.
+    Possible values include: 'Public', 'Private'
```
Comment (Ajnbro) [inline idem_azurerm/exec/azurerm/containerinstance/group.py:184]: I would indent this more so you can more easily tell that this sentence is still a part of the `type` parameter description.

File: idem_azurerm/states/azurerm/containerinstance/group.py [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+
+            **timeout_seconds**: The timeout seconds.
+
+    :param os_type: The operating system type required by the containers in the container group. Possible values
+        include: 'Windows', 'Linux'
+
+    :param restart_policy: Restart policy for all containers within the container group. Possible values are:
+    - ``Always``: Always restart
+    - ``OnFailure``: Restart on failure
+    - ``Never``: Never restart.
```
Comment (Ajnbro) [inline idem_azurerm/states/azurerm/containerinstance/group.py:193]: Remove this period for uniformity

File: idem_azurerm/states/azurerm/containerinstance/group.py [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+        container group.
+
+    :param image_registry_credentials: A list of dictionaries defining ImageRegistryCredential objects for the image
+        registry credentials.
+
+    :param ip_address: A dictionary defining an IpAddress object which represents the IP address for the container
+        group. Possible keys are:
+    - ``ports``: Required if ip_address is used. The list of ports exposed on the container group.
+    - ``type``: Required if ip_address is used. Specifies if the IP is exposed to the public internet or private VNET.
+    Possible values include: 'Public', 'Private'
```
Comment (Ajnbro) [inline idem_azurerm/states/azurerm/containerinstance/group.py:205]: I would indent this more so you can more easily tell that this sentence is still a part of the `type` parameter description.

File: idem_azurerm/exec/azurerm/containerinstance/container.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+            "containerinstance", str(exc), **kwargs
+        )
+        result = {"error": str(exc)}
+
+    if platform.system() == "Windows":
+        _start_exec_pipe_win(result["web_socket_uri"], result["password"])
+    else:
+        _start_exec_pipe(result["web_socket_uri"], result["password"])
+
+    return ""
```
[the remark this answers is not in the document]
Comment (nicholasmhughes) [reply idem_azurerm/exec/azurerm/containerinstance/container.py:234]: that's fair. just needed something that was invisible on the cli. empty dict works too.

File: idem_azurerm/exec/azurerm/containerinstance/group.py (hunk 2 above)
Comment (nicholasmhughes) [reply idem_azurerm/exec/azurerm/containerinstance/group.py:184]: RST formatting these docstrings is the worst...

**Pull request merged**